### ISSUE TITLE
Assertion failure with partition exchange

### ISSUE THREAD

**Issue by hlinnaka** [greenplum-db/gpdb#1517]
Bumped into this while converting the OID consistency TINC tests to installcheck-good:
```
drop table if exists constraint_pt2;
drop table if exists constraint_pt3;
drop table if exists constraint_t2;
drop table if exists constraint_t3;


CREATE TABLE constraint_pt2 (id int, date date, amt decimal(10,2)
       CHECK (amt > 0)) DISTRIBUTED BY (id)
PARTITION BY RANGE (date)
      (PARTITION Jan08 START (date '2008-01-01') INCLUSIVE,
      PARTITION Feb08 START (date '2008-02-01') INCLUSIVE,
      PARTITION Mar08 START (date '2008-03-01') INCLUSIVE
      END (date '2008-04-01') EXCLUSIVE);

CREATE TABLE constraint_pt3 (id int, year int, month int CHECK (month > 0),
       day int CHECK (day > 0), region text)
DISTRIBUTED BY (id)
PARTITION BY RANGE (year)
      SUBPARTITION BY RANGE (month)
      SUBPARTITION TEMPLATE (
      START (1) END (5) EVERY (1),
      DEFAULT SUBPARTITION other_months )
      SUBPARTITION BY LIST (region)
      SUBPARTITION TEMPLATE (
      SUBPARTITION usa VALUES ('usa'),
      SUBPARTITION europe VALUES ('europe'),
      SUBPARTITION asia VALUES ('asia'),
      DEFAULT SUBPARTITION other_regions )
( START (2001) END (2003) EVERY (1),
DEFAULT PARTITION outlying_years );

INSERT INTO constraint_pt3 SELECT i, 2001, 02, i, 'usa' FROM generate_series(1,5)i;
INSERT INTO constraint_pt3 SELECT i, 2001, 02, i, 'europe' FROM generate_series(1,5)i;
INSERT INTO constraint_pt3 SELECT i, 2002, 02, i, 'europe' FROM generate_series(1,5)i;
INSERT INTO constraint_pt3 SELECT i, 2002, 4, i, 'asia' FROM generate_series(1,5)i;
INSERT INTO constraint_pt3 SELECT i, 2002, 4, i, 'europe' FROM generate_series(1,5)i;


INSERT INTO constraint_pt2 SELECT i, '2008-01-13', i FROM generate_series(1,5)i;
INSERT INTO constraint_pt2 SELECT i, '2008-01-20', i FROM generate_series(11,15)i;
INSERT INTO constraint_pt2 SELECT i, '2008-02-13', i FROM generate_series(1,5)i;
INSERT INTO constraint_pt2 SELECT i, '2008-03-13', i FROM generate_series(1,5)i;


CREATE TABLE constraint_t3 (id int, date date, amt decimal(10,2))
       WITH (appendonly=true, orientation=column) DISTRIBUTED BY (id);
INSERT INTO constraint_t3 SELECT i, '2008-03-02', i FROM generate_series(11,15)i;

-- add, rename and exchange
ALTER TABLE constraint_pt2 ADD PARTITION START (date '2009-02-01') INCLUSIVE
       END (date '2009-03-01') EXCLUSIVE,
       EXCHANGE PARTITION mar08 WITH TABLE constraint_t3,
       RENAME PARTITION FOR ('2008-01-16') TO jan2ndhalf;

ALTER TABLE constraint_pt3 ALTER PARTITION FOR ('2001')
      ALTER PARTITION FOR ('2')
      EXCHANGE PARTITION FOR ('europe') WITH TABLE constraint_t3;
```

This leads to assertion failure:
```
psql:part-exchange-assert-fail.sql:53: NOTICE:  CREATE TABLE will create partition "constraint_pt2_1_prt_r1340581185" for table "constraint_pt2"
psql:part-exchange-assert-fail.sql:53: NOTICE:  renamed partition "jan08" to "jan2ndhalf" for relation "constraint_pt2"
ALTER TABLE
psql:part-exchange-assert-fail.sql:57: FATAL:  Unexpected internal error (tablecmds.c:13469)
DETAIL:  FailedAssertion("!(ok)", File: "tablecmds.c", Line: 13469)
```

**Comment by hlinnaka** [comment]
I don't think this has anything to do with the planner - label removed.

**Comment by hlinnaka** [comment]
This was fixed by PR #3066. Was there something remaining to do here? I assume no, so I'll close this, but please re-open if I'm missing something.
